plugins/occt/module/vtkF3DOCCTReader.cxx

```diff
@@ -458,6 +471,40 @@ int vtkF3DOCCTReader::RequestData(
 
   Message::DefaultMessenger()->RemovePrinters(STANDARD_TYPE(Message_PrinterOStream));
 
+  if (this->FileFormat == FILE_FORMAT::BREP)
+  {
+    TopoDS_Shape shape;
+    ProgressIndicator pIndicator(this);
+    const Message_ProgressRange pRange = pIndicator.Start();
+
+    bool success = false;
+    try
+    {
+      success = BinTools::Read(shape, this->GetFileName().c_str(), pRange);
+    }
+    catch (Storage_StreamTypeMismatchError&)
+    {
+      const BRep_Builder builder;
+      success = BRepTools::Read(shape, this->GetFileName().c_str(), builder, pRange);
+    }
+
+    if (success)
+    {
+      output->SetNumberOfBlocks(1);
+      const vtkSmartPointer<vtkPolyData> polydata = this->Internals->CreateShape(shape);
+      if (polydata && polydata->GetNumberOfCells() > 0)
+      {
+        output->SetBlock(1, polydata);
+      }
+      return 1;
+    }
+    else
+    {
+      vtkErrorWithObjectMacro(this, "Failed opening file " << this->GetFileName());
+      return 0;
+    }
```

Comment:
you should be able to test (at least cover it) this easily with a test at VTK module level `TestF3DOCCTReader.cxx`

Reply:
When I try opening the `.brep` file as `STEP` format to trigger the code block the test picks up the `... vtkF3DOCCTReader.cxx:504    ERR| ...` message as a failure.
Do I need a separate `.cxx` file with some CMake magic to expect failure as success?

Reply:
Right, I forgot about that.

> Do I need a separate .cxx file with some CMake magic to expect failure as success?

Hum, we do have some examples of that in the application testing. If you had a invalid.brep (just an empty file) and try opening it with F3D, do you hit this code ?

If that is so, you can copy the `TestInvalidFullScene` test, where the test result does not matter as long as the text is present.

Reply:
empty file seems to do the trick, thanks!